pom.xml

```diff
@@ -150,6 +154,34 @@
         </executions>
       </plugin>
 
+      <plugin>
+        <artifactId>maven-jar-plugin</artifactId>
+        <executions>
+          <execution>
+            <!-- Produce a jar with a platfrom-specific classifier. -->
+            <id>default-jar</id>
+            <goals>
+              <goal>jar</goal>
+            </goals>
+            <phase>package</phase>
+            <configuration>
+              <classifier>${os.detected.classifier}</classifier>
+            </configuration>
+          </execution>
+          <execution>
+            <!-- Also produce a jar with 'java' classifier. This can be built on any platform. -->
```

Comment:
Under this model, the non-classified jar is produced by the shade pom to produce a jar that can be used on any platform. Here, the antrun plugin produces the platform-specific jar. This extra execution tells maven to package up the normal target results as a 'java' package.